pkg/client/client.go

```diff
@@ -261,3 +262,54 @@ func MergeAnnotations(toolchainObject client.Object, newAnnotations map[string]s
 	}
 	toolchainObject.SetAnnotations(annotations)
 }
+
+// ApplyUnstructuredObjects applies the given Unstructured objects on the cluster.
+func ApplyUnstructuredObjects(ctx context.Context, cl client.Client, unstructuredObjects []*unstructured.Unstructured, newLabels map[string]string) (bool, error) {
+	applyClient := NewApplyClient(cl)
+	anyApplied := false
+
+	for _, unstructuredObj := range unstructuredObjects {
+		var object = unstructuredObj.DeepCopyObject().(client.Object)
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredObj.Object, object); err != nil {
+			return false, err
+		}
+		// Special handling of ServiceAccounts is required because if a ServiceAccount is reapplied when it already exists, it causes Kubernetes controllers to
+		// automatically create new Secrets for the ServiceAccounts. After enough time the number of Secrets created will hit the Secrets quota and then no new
+		// Secrets can be created. To prevent this from happening, we fetch the already existing SA, update labels and annotations only, and then call update using the same object (keeping the refs to secrets).
+		if strings.EqualFold(object.GetObjectKind().GroupVersionKind().Kind, "ServiceAccount") {
```

Comment:
do you think that we could move it directly to the `applyObject` function?

Reply:
Good point, I've implemented the functionality in `applyObject` in https://github.com/codeready-toolchain/toolchain-common/pull/373/commits/8ee46aacee2e9c2a0bb3a24f7c0027556cc34dfc . 

It's not using the existing object but just copying the secrets and annotations, please take a look and let me know if I've missed anything.